src/impl/realm_coordinator.cpp

```diff
@@ -279,6 +280,37 @@ void RealmCoordinator::do_get_realm(Realm::Config config, std::shared_ptr<Realm>
 
     realm_lock.unlock();
     if (schema) {
+#if REALM_ENABLE_SYNC && ANDROID
```

Comment:
So it will also be enabled for React Native, right?

Reply:
I suspect so, I don't think we have defines for each binding? Guess I'll need to add that.

Reply:
In Realm JS we already have `REALM_PLATFORM_NODE` so I could add the flags  `REALM_PLATFORM_REACTNATIVE_ANDROID` and `REALM_PLATFORM_REACTNATIVE_IOS`.

Reply:
Wouldn't it make more sense that I added it to Java so it would become:

```
#if REALM_ENABLE_SYNC && REALM_PLATFORM_JAVA
```

That way we also exclude Xamarin for Android?